File: StructuralPattern/Composite/Java/Concrete/Folder.java
package Concrete;

import Interface.FileSystemComponent;
import java.util.ArrayList;
import java.util.List;

public class Folder implements FileSystemComponent {
    // Variables of the folder
    private String name;
    private List<FileSystemComponent> children = new ArrayList<FileSystemComponent>();
    // TODO: add more variables

    // Constructor
    public Folder(String name) {
        this.name = name;
    }

    // Add a child to the folder
    public void add(FileSystemComponent child) {
        children.add(child);
    }

    // Remove a child from the folder
    public void remove(FileSystemComponent child) {
        children.remove(child);
    }

    // Show details about the folder
    @Override
    public void showDetails() {
        System.out.println("Folder: " + name);
        for (FileSystemComponent child : children) {
            child.showDetails();
        }
    }

    // Getter and setter
    // TODO ...
}

File: StructuralPattern/Composite/Python/Interface/Folder.py

from Concrete import FileSystemComponent

class Folder(FileSystemComponent):
    """
        Class Folder
    """
    def __init__(self, name):
        """
            Constructor
        """
        self.name = name
        self.files = []
        super().__init__(name)
    
    def addFile(self, file):
        """
            Add file
        """
        self.files.append(file)
        
    def removeFile(self, file):
        """
            Remove file
        """
        self.files.remove(file)
    
    def getDetails(self):
        """
            Show all the details
        """
        print("Folder: " + self.name)
        for file in self.files:
            file.getDetails()